### ISSUE TITLE
Password strength indicator is inaccurate if using UUID references

### ISSUE THREAD

**Issue by ebkalderon** [keepassxreboot/keepassxc#10295]
## Overview

It seems that the password strength indicator bar doesn't correctly account for password references (i.e. `{REF:P@I:$uuid}`) created as a result of a "Clone Entry..." operation.

The password strength indicator seems to be evaluating the `{REF:P@I:$uuid}` string _literally_ instead of the password that it points to, and this always results in a 100% full green bar (indicating "maximum strength"). This misleads a casual user into assuming the password for this entry is strong, when the source password in reality may be much weaker.

## Steps to Reproduce

1. Create a new KDBX database.
2. Create a new entry named "test" with a weak password. Observe the red-colored "weak" password indicator. (_Figure 1_)
3. Clone the "test" entry with <kbd>Ctrl+K</kbd>. **Be sure to check "Replace username and password with references".** (_Figure 2_)
4. Edit the "test - Clone" entry. Observe the green "maximum strength" password indicator. (_Figure 3_)

<details> <summary>Figure 1</summary>

![Screenshot from 2024-02-11 14-30-41](https://github.com/keepassxreboot/keepassxc/assets/322831/ab570451-70e9-421b-b786-440ce970464a)

</details>

<details> <summary>Figure 2</summary>

![Screenshot from 2024-02-11 14-31-16](https://github.com/keepassxreboot/keepassxc/assets/322831/97484bb7-e169-4c1a-aa41-24e3ec3fa412)

</details>

<details> <summary>Figure 3</summary>

![Screenshot from 2024-02-11 14-31-32](https://github.com/keepassxreboot/keepassxc/assets/322831/60053557-5b2e-437b-8cc7-d29a584b0359)

</details>

## Expected Behavior

The password strength indicator for both the original "test" entry and the "test - Clone" entry should be identical (both entries' indicators should be red; it's a weak password).

## Actual Behavior

The password strength indicator for the original "test" entry and the "test - Clone" entry are different. The strength indicator for the cloned entry seems to indicate the original password is strong, when it really isn't.

## Context

KeePassXC - Version 2.7.6
Revision: dd21def

Operating System: Linux
Desktop Env: GNOME
Windowing System: Wayland

**Comment by droidmonkey** [comment]
Good find!

**Comment by ebkalderon** [comment]
Thanks, @droidmonkey! It's not a huge problem, but it would be nice if KeePassXC exhibited the more intuitive expected behavior here.

Also, I couldn't find any existing tickets that described this exact issue, but maybe my search-fu wasn't strong enough. Feel free to close this ticket as duplicate if you happen to find an existing one, provided that your link two together before closing.

**Comment by droidmonkey** [comment]
I don't think this is a duplicate, might be a tad challenging to fix though since resolving references requires access to a database object which we don't expose to the password widget.
